### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Zelinskyi-Serhii**
[DEMO LINK](https://Zelinskyi-Serhii.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by Yevhenii0536 — APPROVED** [review]
gj!

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,136 @@
...
+      reset();
+    }
+
+    if (!userId) {
+      setErrorUser(true);
+    }
+
+    if (!title.trim()) {
+      setErrorTitle(true);
+    }
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:62]:
```suggestion
    
    if (!userId) {
      setErrorUser(true);
      
      return;
    }

    if (!title.trim()) {
      setErrorTitle(true);
      
      return;
    }

    const newTodo = {
      id: findMax(todos),
      title,
      completed: false,
      userId,
      user: getUser(userId),
    };

    addTodos([...todos, newTodo]);
    reset();
```

Comment (OleksiiBlezniuk) [reply src/App.tsx:62]: Your first check - is code that will run if other checks will failed

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,136 @@
...
+            <input
+              type="text"
+              data-cy="titleInput"
+              id="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={(event) => {
+                setTitle(event.target.value);
+                setErrorTitle(false);
+              }}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:86]: Let's create handler

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,136 @@
...
+            />
+          </label>
+
+          {errorTitle && <span className="error">Please enter a title</span>}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:90]:
```suggestion
          {errorTitle && (
            <span className="error">
              Please enter a title
            </span>
          )}
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,136 @@
...
+          <label htmlFor="userSelect">
+            {'User: '}
+            <select
+              data-cy="userSelect"
+              id="userSelect"
+              value={userId}
+              onChange={(event) => {
+                addUserId(+event.target.value);
+                setErrorUser(false);
+              }}
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:103]: Let's create handler

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]
Let's add some styles to the page

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,143 @@
...
+    setErrorTitle(false);
+    setErrorUser(false);
+  };
+
+  const addTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setErrorUser(!userId);
+    setErrorTitle(!title);
+
+    if (userId && title.trim()) {
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:47]: I guess will be more logical to change this check a bit

Comment (OleksiiBlezniuk) [inline src/App.tsx:47]:
```suggestion
    if (!userId || !title) {
      setErrorUser(!userId);
      setErrorTitle(!title);
      
      return;
    }
    
```

File: src/App.tsx [hunk 6]
```diff
@@ -1,61 +1,143 @@
...
+
+  const handleAddTitle = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTitle(event.target.value);
+    setErrorTitle(false);
+  };
+
+  const handleAddUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    addUserId(+event.target.value);
```
Comment (OleksiiBlezniuk) [inline src/App.tsx:69]:
```suggestion
    addUserId(Number(event.target.value));
```

**Review by OleksiiBlezniuk — APPROVED** [review]
Great that you changed initial UI, good job 🔥 
I suggest it will be better to use lighter color for title or add some light stroke or shadow for text to make it more noticable on dark gradient

**Pull request closed without merging**